src/ledger.rs

```diff
@@ -89,9 +91,17 @@ fn entry_at<A: Read + Seek>(file: &mut A, at: u64) -> io::Result<Entry> {
     read_entry(file, len)
 }
 
-fn next_entry<A: Read>(file: &mut A) -> io::Result<Entry> {
+fn next_entry<A: Read>(file: &mut A, cur_pos: &mut usize) -> io::Result<Entry> {
     let len = deserialize_from(file.take(SIZEOF_U64)).map_err(err_bincode_to_io)?;
-    read_entry(file, len)
+    let result: io::Result<Entry> = read_entry(file, len);
```

Comment:
This type annotation shouldn't be needed.

Reply:
Also, the Rust idiom here is the `?` operator. Looks like this code can be reduced to some small variation of:

```rust
let entry = read_entry(file, len)?;
*cur_pos += SIZEOF_U64 + len;
entry
```